lsp/nls/src/linearization/completed.rs

```diff
@@ -40,14 +34,12 @@ impl Completed {
 
     pub fn get_in_scope(
         &self,
-        LinearizationItem { scope, .. }: &LinearizationItem<Resolved>,
+        LinearizationItem { env, .. }: &LinearizationItem<Resolved>,
     ) -> Vec<&LinearizationItem<Resolved>> {
-        let empty = Vec::with_capacity(0);
-        (0..scope.len())
-            .flat_map(|end| self.scope.get(&scope[..=end]).unwrap_or(&empty))
-            .map(|id| self.get_item(*id))
-            .flatten()
-            .collect()
+        env.iter()
+            .map(|(_, id)| self.get_item(*id))
+            .collect::<Option<Vec<_>>>()
+            .unwrap_or(Vec::new())
```

Comment:
FYI, there is a slight variation function for that:

```suggestion
            .unwrap_or_default()
```

Reply:
Nice, thanks!